cmd/monaco/deploy/deploy.go

```diff
@@ -282,15 +278,15 @@ func validateAuthenticationWithProjectConfigs(projects []project.Project, enviro
 						}
 					case config.SettingsType:
 						t, ok := conf.Type.(config.SettingsType)
-						if ok && t.AllUserPermission != nil && environments[envName].Auth.OAuth == nil {
-							return fmt.Errorf("using permission property on settings API requires OAuth, schema '%s' enviroment '%s'", t.SchemaId, envName)
+						if ok && t.AllUserPermission != nil && !environments[envName].HasPlatformAuth() {
+							return fmt.Errorf("using permission property on settings API requires platform authentication, schema '%s' enviroment '%s'", t.SchemaId, envName)
 						}
-						if environments[envName].Auth.ApiToken == nil && environments[envName].Auth.OAuth == nil {
-							return fmt.Errorf("API of type '%s' requires an API token or OAuth for environment '%s'", conf.Type, envName)
+						if environments[envName].Auth.ApiToken == nil && !environments[envName].HasPlatformAuth() {
+							return fmt.Errorf("API of type '%s' requires an API token or platform authentication for environment '%s'", conf.Type, envName)
 						}
 					default:
-						if environments[envName].Auth.OAuth == nil {
-							return fmt.Errorf("API of type '%s' requires OAuth for environment '%s'", conf.Type, envName)
+						if !environments[envName].HasPlatformAuth() {
+							return fmt.Errorf("API of type '%s' requires platform authentication for environment '%s'", conf.Type, envName)
```

Comment:
As we're using the term platform a lot, we should probably tighten the language here:
- "platform authentication" vs "platform credentials"

And also I notice:
- "API of type ..." vs "... resources"